### PR TITLE
removing is_commutative

### LINKED ISSUE #23721: IndexedBase non-commutivity fails
Hi there, 

I believe I've found a bug relating to the IndexedBase class, where it doesn't seem to respect non-commutivity of objects.  If I have two non-commuting IndexedBase objects `t` and `u`, I would expect that `t[0]*u[0]+u[0]*t[0]!=2*t[0]*u[0]`.  The below code highlights the issue, showing that all the objects have `commutivity=False`, yet the addition of `t[0]*u[0]` with `u[0]*t[0]` simplified to `2t[0]u[0]`

    import sympy as sp                                                                                  
                                                                                                    
    t=sp.IndexedBase('t',commutative=False)                                                             
    u=sp.IndexedBase('u',commutative=False)                                                             
                                                                                                      
    print('t commutativity is {}'.format(t._assumptions['commutative']))                                    
    print('t[0] commutativity is {}'.format(t[0]._assumptions['commutative']))                           
    print('u commutativity is {}'.format(u._assumptions['commutative']))                                    
    print('u[0] commutativity is {}'.format(u[0]._assumptions['commutative']))                          
                                                                                                    
    print(t[0]*u[0]+u[0]*t[0])    
    
The output I get in sympy version 1.10.1 is 

    t commutativity is False
    t[0] commutativity is False
    u commutativity is False
    u[0] commutativity is False
    2*t[0]*u[0]

### PR DESCRIPTION
**Pull request by harshkasat**
#### References to other Issues or PRs

Fixes: #23721

#### Release Notes

* tensor
  *  Fixed the issue that `commutative=False` assumption is not working with `IndexedBase`. For example, if `t` and `u` are `IndexedBase` with `commutative=False`, `u[0]*t[0]` should not simplify to `u[0]*t[0]`.

### CODE REVIEW

File: sympy/tensor/tests/test_indexed.py [hunk 1]
```diff
@@ -501,3 +501,10 @@ def test_complicated_derivative_with_Indexed():
             ((x[i] - y[i])**2/sigma,)
         )/sigma**2
     )
+
+
+def test_IndexedBase_commutative():
+    t = IndexedBase('t' ,commutative=False)
+    u = IndexedBase('u' ,commutative=False)
```
Comment (sylee957) [inline sympy/tensor/tests/test_indexed.py:508]:
```suggestion
    t = IndexedBase('t', commutative=False)
    u = IndexedBase('u', commutative=False)
```

**Comment by sylee957** [comment]
Please write more detailed and correct release note. I think that `removing is_commutative` is too broad and ambiguous, and sounds scary like every `is_commutative` is removed from SymPy.

**Comment by harshkasat** [comment]
@sylee957 can you checkout a pr. If any change required

**Comment by sylee957** [comment]
Unfortunately, I still find the release note difficult to follow.

Also, the merge could be broken because I see unrelated contributors tagged. https://github.com/sympy/sympy/pull/26051#issuecomment-1880161169 This may also need to be fixed

**Pull request merged**